src/org/sparkboard/client/auth.cljs

```diff
@@ -0,0 +1,47 @@
+(ns org.sparkboard.client.auth
+  (:require ["react" :as react]
+            [triple.view :as v]
+            [applied-science.js-interop :as j]
+            [org.sparkboard.client.firebase :as firebase]
+            [kitchen-async.promise :as p]))
+
+(v/defview after-promise [{:keys [fallback promise]} element]
+  (let [mounted-ref (react/useRef true)
+        [done? done!] (react/useState nil)]
+    (react/useEffect (fn []
+                       (p/do promise
+                             (when (true? (j/get mounted-ref :current))
+                               (done! true)))
+                       (fn [] (j/assoc! mounted-ref :current false))))
+    (if done?
+      element
+      fallback)))
+
+(v/defview use-firebaseui-web []
+  (react/useEffect
+    (fn []
+      (j/call @firebase/UI :start "#firebaseui" firebase/ui-config)
+      js/undefined))
+  [:div#firebaseui])
+
+(v/defview auth-header* []
+  (let [auth-state @firebase/current-user]
+    (cond (= :initial-state auth-state) "..."
```

Comment:
I don't understand the `"..."` here, or `".."` on line 46. Are these actual values from Firebase, or a TODO?

Reply:
This is just displaying a ".." and then "..." (rather quickly) on page-load as the auth flow progresses. Should probably be replaced with a loading animation.